### PR TITLE
Writing a readme for jsonz.

### PR DESCRIPTION
**Pull request by adamdecaf**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -1,10 +1,163 @@
...
+  val json = """{"thing": 1}"""
+  Jsonz.fromJsonStr[Foo](json)
+}
+```
+
+## Failures
+
+Jsonz is different about failures. When parsing json failures are accumulated and delivered at the end. The type `JsonzValidation[T]` is a type alias for scalaz's `Validation[NonEmptyList[JsFailure], T]` so the user will receive either a `scalaz.Success` with an instance of the model filled in with the data from the json, or a `scalaz.Failure` that contains failures that occured duirng parsing. These failures could be from missing fields, or malformly typed elements.
+
+### No Exceptions
```
Comment (trentio) [inline README.markdown]: Im not sure that this section is actually needed.  The fact that it returns the failures in a NEL would imply that the library doesn't blow up while parsing.

Comment (adamdecaf) [inline README.markdown]: I wasn't sure if it did imply that. But some other exceptions are caught that I didn't explicitly list.

File: README.markdown [hunk 2]
```diff
@@ -1,10 +1,163 @@
...
+```scala
+import jsonz._
+import jsonz.DefaultFormats._
+
+object Example {
+    case class Foo(thing: Int, other: String)
+    implicit val fooFormat = productFormat2("thing", "other")(Foo.apply)(Foo.unapply)
+
+    Jsonz.fromJsonStr[Foo]("""{"thing":12, "other":"hi"}""") // JsonzValidation[Foo]
+    Jsonz.toJsonStr(Foo(12, "hi")) // JsValue
```
Comment (trentio) [inline README.markdown]: This would actually return a string

File: README.markdown (hunk 1 above)
Comment (DustinChaloupka) [inline README.markdown]: I don't know much about other libraries, but if the standard is to throw exceptions I think this is a nice tidbit to add and recognize.

File: README.markdown [hunk 3]
```diff
@@ -1,10 +1,163 @@
...
+### Joda-Time
+
+Joda-Time is typically the first library chosen for performing date/time operations on the JVM. Jsonz offers support for reading and writing it's `org.joda.time.DateTime` class via the following imports. The formats used to try and parse are: `ISO 8601` (with and without millis), `yyyy-MM-dd`, and [RFC 1123](https://tools.ietf.org/html/rfc1123) (EEE, dd MMM yyyy HH:mm:ss 'GMT').
+
+```scala
+import jsonz._
+import jsonz.joda.JodaTimeFormats._
+```
+
+### Spray
```
Comment (trentio) [inline README.markdown]: Isnt spray a provided dependency, that should probably be noted.

**Comment by DustinChaloupka** [comment]
https://camo.githubusercontent.com/b3e29ddc23d86c93c8faf97e69484e64d3cad175/687474703a2f2f692e696d6775722e636f6d2f523779384364352e676966

File: README.markdown [hunk 4]
```diff
@@ -1,10 +1,163 @@
...
+Given a recursive type you will need to wrap the format in the `lazyFormat` method.
+
+```scala
+import jsonz._
+import jsonz.DefaultFormats._
+case class RecursiveType(rs: List[RecursiveType])
+implicit lazy val RecursiveTypeFormat: Format[RecursiveType] = lazyFormat(productFormat1("rs")(RecursiveType.apply)(RecursiveType.unapply))
+```
+
+### Joda-Time
```
Comment (trentio) [inline README.markdown]: Also a provided dependency

File: README.markdown [hunk 5]
```diff
@@ -1,10 +1,163 @@
...
+### Spray
+
+Spray versions 1.1.x, 1.2.x, and 1.3.x are supported to read entit ies and complete requests as json via jsonz. To use the following import will need to be used, or to mixin the associated trait under the same name.
+
+```scala
+import jsonz.spray.JsonzMarshalling._ // companion object
+import jsonz.spray.JsonzMarshalling   // trait
+```
+
+### Specs2
```
Comment (trentio) [inline README.markdown]: One more provided dependency.

File: README.markdown (hunk 1 above)
Comment (zmccoy) [inline README.markdown]: I think it's okay to keep it in here.  It's a small note that just states that the library does this.  Someone that is not familiar with NEL can read this and understand what is going without having to infer it from the types that we use.

Comment (trentio) [inline README.markdown]: What other exceptions are caught, you only list the ones that are rolled into the Failure

File: README.markdown (hunk 3 above)
Comment (adamdecaf) [inline README.markdown]: Noted.

File: README.markdown (hunk 1 above)
Comment (adamdecaf) [inline README.markdown]: Jackson's ParsingException, and any customly defined JsFailure statements.

**Comment by zmccoy** [comment]
:+1:

**Comment by rubbish** [comment]
This all looks good to me. Thanks @adamdecaf

**Pull request merged**